instance/tests/models/test_openedx_database_mixins.py

```diff
@@ -0,0 +1,285 @@
+        base_domain = 'this-is-a-really-long-unusual-domain-แปลกมาก.com'
+        self.instance = OpenEdXInstanceFactory(use_ephemeral_databases=False,
+                                               sub_domain=sub_domain,
+                                               base_domain=base_domain)
+        self.instance.provision_mysql()
+        self.check_mysql()
+
+    def test_provision_mysql_again(self):
+        """
+        Only create the database once
+        """
+        self.instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        self.instance.provision_mysql()
+        self.assertIs(self.instance.mysql_provisioned, True)
+
+        mysql_user = self.instance.mysql_user
+        mysql_pass = self.instance.mysql_pass
+        self.instance.provision_mysql()
+        self.assertEqual(self.instance.mysql_user, mysql_user)
+        self.assertEqual(self.instance.mysql_pass, mysql_pass)
+        self.check_mysql()
+
+    @patch_services
+    @override_settings(INSTANCE_MYSQL_URL_OBJ=urlparse('mysql://user:pass@mysql.opencraft.com'))
+    def test_ansible_settings_mysql(self, mocks):
+        """
+        Add mysql ansible vars if INSTANCE_MYSQL_URL is set and not using ephemeral databases
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        instance.provision_mysql()
+
+        db_vars = yaml.load(instance.get_database_settings())
+        self.assertGreater(len(instance.mysql_user), 5)
+        self.assertGreater(len(instance.mysql_pass), 10)
+        self.assertEqual(db_vars['EDXAPP_MYSQL_USER'], instance.mysql_user)
+        self.assertEqual(db_vars['EDXAPP_MYSQL_PASSWORD'], instance.mysql_pass)
+        self.assertEqual(db_vars['EDXAPP_MYSQL_HOST'], 'mysql.opencraft.com')
+        self.assertEqual(db_vars['EDXAPP_MYSQL_PORT'], 3306)
+        self.assertEqual(db_vars['EDXAPP_MYSQL_DB_NAME'], instance.mysql_database_name)
+        self.assertEqual(db_vars['COMMON_MYSQL_MIGRATE_USER'], instance.mysql_user)
+        self.assertEqual(db_vars['COMMON_MYSQL_MIGRATE_PASS'], instance.mysql_pass)
+
+    @patch_services
+    @override_settings(INSTANCE_MYSQL_URL_OBJ=None)
+    def test_ansible_settings_mysql_not_set(self, mocks):
+        """
+        Don't add mysql ansible vars if INSTANCE_MYSQL_URL is not set
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=False)
+        instance.provision_mysql()
+        self.check_mysql_vars_not_set(instance)
+
+    @patch_services
+    @override_settings(INSTANCE_MYSQL_URL_OBJ=urlparse('mysql://user:pass@mysql.opencraft.com'))
+    def test_ansible_settings_mysql_ephemeral(self, mocks):
+        """
+        Don't add mysql ansible vars for ephemeral databases
+        """
+        instance = OpenEdXInstanceFactory(use_ephemeral_databases=True)
+        instance.provision_mysql()
+        self.check_mysql_vars_not_set(instance)
+
+    def check_mysql_vars_not_set(self, instance):
```

Comment:
@bradenmacdonald Nit: Move above `test_*` methods (below `check_mysql`) to keep helper methods and test methods separate.